class Solution {
    public int minInsertions(String s) {
        int open = 0;
        int ans = 0;
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (ch == '(') {
                open++;
            } else {
                if (i + 1 < s.length() && s.charAt(i + 1) == ')') {
                    i++;
                } else {
                    ans++; // Insert one ')'
                }
                if (open > 0) {
                    open--;
                } else {
                    ans++; // Insert one '('
                }
            }
        }
        return ans + 2 * open;
    }
}